src/db/seeds.rb

```diff
@@ -8,9 +8,20 @@
 
 logger.info 'Creating functions...'
 functions = users.map do |user|
-  FactoryBot.create_list(:function, 10, user: user)
+  FactoryBot.create_list(:function, 10, user: user, tags: [])
```

Comment:
instead just do this 
```suggestion
  FactoryBot.create_list(:function, 10, user: user, tags_list: ['general', 'string', 'image', 'numeric', 'rails'].sample(3))
```

your tags_list= method should handle the rest, you don't need the changes from 14 to 24 

your model setter should

Reply:
`tags_list` isn't an array its a string and it's serialized in the model setter. So instead of generating an array of strings then joining them as a string I decided that it's better to directly set the `tags` property. I think this a better approach than directly hardcoding them.